### DISCUSSION TITLE
Configuration file

### DISCUSSION THREAD

**Discussion by daidaiV5** [nextflow-io/nextflow#2232] [Q&A]
Hello
I have a question about Configuration file.
by default, the nextflow will use config file named nextflow.config in the current directory, can we change the name of nextflow.config and the nextflow will also use this file by default. in other words,  I would like to use Config file specified (name is not nextflow.config ) by default instead of using the -c my_config option every time.
Thanks for taking the time to read.
Yuping

**Comment by pditommaso** [comment]
Hello, it should be possible to control via the `NXF_CONFIG_FILE` env variable

**Reply by daidaiV5** [reply]
Thank you for your quick response, if I change env variable NXF_CONFIG_FILE , all files will use the Config file specified by default ? can I use one config file special for one main.nf, for example, main1.nf will use the main1.config by default and the main2.nf will use main2.config by default.

**Reply by pditommaso** [reply]
when setting `NXF_CONFIG_FILE=foo` nextflow look for a config named `foo` instead of `nextflow.config`
